src/AplosConnector.Common/Services/AplosIntegrationService.cs

```diff
@@ -1203,6 +1207,7 @@ private async Task<List<TransactionModel>> SyncTransactions(
 
                             if (!string.IsNullOrEmpty(syncIneligibilityReason))
                             {
+                                failureCount++;
```

Comment:
this is the key fix. the rest is basically adding transaction id to the various `syncIneligibilityReason` and instead of logging (it will be logged later), setting `syncIneligibilityReason` when it's a "failure" so it's counted due to the above fix.